feat(core): add packageManager field when setting up new workspaces

juristr opened the pull request:
<!-- Please make sure you have read the submission guidelines before posting an PR -->
<!-- https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#-submitting-a-pr -->

<!-- Please make sure that your commit message follows our format -->
<!-- Example: `fix(nx): must begin with lowercase` -->

<!-- If this is a particularly complex change or feature addition, you can request a dedicated Nx release for this pull request branch. Mention someone from the Nx team or the `@nrwl/nx-pipelines-reviewers` and they will confirm if the PR warrants its own release for testing purposes, and generate it for you if appropriate. -->

## Current Behavior
<!-- This is the behavior we have today -->

Currently we don't add the `packageManager` property when generating a new workspace. 

## Expected Behavior
<!-- This is the behavior we should expect with the changes in this PR -->

We should have it as more and more tooling relies on it to make decisions. (e.g. on CI etc..)

## Related Issue(s)
<!-- Please link the issue being fixed so it gets closed when this is merged. -->

Fixes #

JamesHenry reviewed (changes requested):
@juristr I think we'll want a custom snapshot serializer for these otherwise they will be subject to change every time our CI environment changes by even a patch version of npm

juristr commented:
@JamesHenry oh yeah good catch

juristr committed:
feat(core): add packageManager field when setting up new workspaces

juristr committed:
fix(core): disable corepack project field enforcement in e2e tests

Prevents corepack from enforcing the packageManager field in workspaces
generated during e2e tests, which causes version mismatch failures on CI.
